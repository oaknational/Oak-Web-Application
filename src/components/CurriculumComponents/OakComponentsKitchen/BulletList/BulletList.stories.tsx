import { OakThemeProvider, oakDefaultTheme } from "@oaknational/oak-components";
import { Meta, StoryObj } from "@storybook/react";

import Component from ".";

const meta: Meta<typeof Component> = {
  component: Component,
  argTypes: {},
};

export default meta;
type Story = StoryObj<typeof Component>;

const TestComponent = () => {
  return (
    <OakThemeProvider theme={oakDefaultTheme}>
      <div>
        <Component
          items={[{ text: "one" }, { text: "two" }, { text: "three" }]}
        />
      </div>
      <div>
        <Component items={[{ text: "single" }]} />
      </div>
    </OakThemeProvider>
  );
};

export const BulletList: Story = {
  render: TestComponent,
};
