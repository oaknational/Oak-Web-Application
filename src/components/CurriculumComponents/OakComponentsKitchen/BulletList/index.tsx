import { OakTypography, OakTypographyProps } from "@oaknational/oak-components";
import { Fragment } from "react";

type BulletListProps = {
  items: { text: string }[];
} & OakTypographyProps;
export default function BulletList(props: BulletListProps) {
  const { items, ...fontProps } = props;

  const children = items.map(({ text }, i) => (
    <Fragment key={`${text}`}>
      <OakTypography {...fontProps}>{text}</OakTypography>
      {i + 1 < items.length && (
        <OakTypography {...fontProps} aria-hidden>
          â€¢
        </OakTypography>
      )}
    </Fragment>
  ));

  return <Fragment>{children}</Fragment>;
}
