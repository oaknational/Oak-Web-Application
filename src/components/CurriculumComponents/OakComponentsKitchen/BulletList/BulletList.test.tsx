import BulletList from ".";

import renderWithTheme from "@/__tests__/__helpers__/renderWithTheme";

describe("BulletList", () => {
  test("render", async () => {
    const { baseElement } = renderWithTheme(
      <>
        <BulletList
          items={[{ text: "one" }, { text: "two" }, { text: "three" }]}
        />
        <BulletList items={[{ text: "single" }]} />
      </>,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
