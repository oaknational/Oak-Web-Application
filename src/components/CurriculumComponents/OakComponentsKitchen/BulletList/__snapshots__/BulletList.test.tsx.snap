// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulletList render 1`] = `
<body>
  <div>
    <div
      class="sc-fqkvVR sc-eldPxv bjeqlm jsxtWU"
    >
      one
    </div>
    <div
      aria-hidden="true"
      class="sc-fqkvVR sc-eldPxv bjeqlm jsxtWU"
    >
      •
    </div>
    <div
      class="sc-fqkvVR sc-eldPxv bjeqlm jsxtWU"
    >
      two
    </div>
    <div
      aria-hidden="true"
      class="sc-fqkvVR sc-eldPxv bjeqlm jsxtWU"
    >
      •
    </div>
    <div
      class="sc-fqkvVR sc-eldPxv bjeqlm jsxtWU"
    >
      three
    </div>
    <div
      class="sc-fqkvVR sc-eldPxv bjeqlm jsxtWU"
    >
      single
    </div>
  </div>
</body>
`;
